src/uk/gov/hmcts/contino/CommonPipelineDsl.groovy

```diff
@@ -0,0 +1,29 @@
+package uk.gov.hmcts.contino
+
+abstract class CommonPipelineDsl implements Serializable {
```

Comment:
what would you think about extracting out the refactoring first and using it in withPipeline?
i.e. this change hasn't been applied to withPipeline, would probably make sense to refactor before this change...

Reply:
yep - kind of did it back-to-front. Started with the intention of refactoring first bet ended up writing the new implementation with the intention of refactoring existing pipeline later. Will do as a separate PR after this change